ark/jsonschema/__tests__/string.test.ts

```diff
@@ -0,0 +1,79 @@
+import { attest, contextualize } from "@ark/attest"
+import { parseJsonSchema } from "@ark/jsonschema"
+
+contextualize(() => {
+	it("type string", () => {
+		const t = parseJsonSchema({ type: "string" })
+		attest(t.json).snap({ domain: "string" })
+	})
+
+	it("maxLength (positive)", () => {
+		const tMaxLength = parseJsonSchema({
+			type: "string",
+			maxLength: 5
+		})
+		attest(tMaxLength.json).snap({
+			domain: "string",
+			maxLength: 5
+		})
+	})
+
+	it("maxLength (negative)", () => {
+		const maxLength = -5
+		attest(() =>
+			parseJsonSchema({
+				type: "string",
+				maxLength
+			})
+		).throws(
+			`TraversalError: maxLength must be non-negative (was ${maxLength})`
+		)
+	})
+
+	it("minLength (positive)", () => {
+		const tMinLength = parseJsonSchema({
+			type: "string",
+			minLength: 5
+		})
+		attest(tMinLength.json).snap({
+			domain: "string",
+			minLength: 5
+		})
+	})
+
+	it("minLength (negative)", () => {
+		const minLength = -1
+		attest(() =>
+			parseJsonSchema({
+				type: "string",
+				minLength
+			})
+		).throws(
+			`TraversalError: minLength must be non-negative (was ${minLength})`
```

Comment:
same as above

Reply:
Marking as resolved. Can be further discussed on https://github.com/arktypeio/arktype/pull/1159#discussion_r2015124232